src/theory/sets/theory_sets_private.cpp

```diff
@@ -404,142 +276,49 @@ bool TheorySetsPrivate::assertFact( Node fact, Node exp ){
-    Trace("sets-assertion") << "(assert (=> " << exp << " " << fact << ")) ; by " << c << std::endl;
-  }
-}
-
-void TheorySetsPrivate::assertInference( Node fact, std::vector< Node >& exp, std::vector< Node >& lemmas, const char * c, int inferType ){
-  Node exp_n = exp.empty() ? d_true : ( exp.size()==1 ? exp[0] : NodeManager::currentNM()->mkNode( kind::AND, exp ) );
-  assertInference( fact, exp_n, lemmas, c, inferType );
-}
-
-void TheorySetsPrivate::assertInference( std::vector< Node >& conc, Node exp, std::vector< Node >& lemmas, const char * c, int inferType ){
-  if( !conc.empty() ){
-    Node fact = conc.size()==1 ? conc[0] : NodeManager::currentNM()->mkNode( kind::AND, conc );
-    assertInference( fact, exp, lemmas, c, inferType );
+  else
+  {
+    return false;
   }
 }
-void TheorySetsPrivate::assertInference( std::vector< Node >& conc, std::vector< Node >& exp, std::vector< Node >& lemmas, const char * c, int inferType ) {
-  Node exp_n = exp.empty() ? d_true : ( exp.size()==1 ? exp[0] : NodeManager::currentNM()->mkNode( kind::AND, exp ) );
-  assertInference( conc, exp_n, lemmas, c, inferType );
-}
 
-void TheorySetsPrivate::split( Node n, int reqPol ) {
-  n = Rewriter::rewrite( n );
-  Node lem = NodeManager::currentNM()->mkNode( kind::OR, n, n.negate() );
-  std::vector< Node > lemmas;
-  lemmas.push_back( lem );
-  flushLemmas( lemmas );
-  Trace("sets-lemma") << "Sets::Lemma split : " << lem << std::endl;
-  if( reqPol!=0 ){
-    Trace("sets-lemma") << "Sets::Require phase "  << n << " " << (reqPol>0) << std::endl;
-    d_external.getOutputChannel().requirePhase( n, reqPol>0 );
-  }
+void TheorySetsPrivate::fullEffortReset()
+{
+  Assert(d_equalityEngine.consistent());
+  d_full_check_incomplete = false;
+  d_most_common_type.clear();
+  d_most_common_type_term.clear();
+  d_card_enabled = false;
+  d_rels_enabled = false;
+  // reset the state object
+  d_state.reset();
+  // reset the inference manager
+  d_im.reset();
+  // reset the cardinality solver
+  d_cardSolver->reset();
 }
 
 void TheorySetsPrivate::fullEffortCheck(){
   Trace("sets") << "----- Full effort check ------" << std::endl;
   do{
     Trace("sets") << "...iterate full effort check..." << std::endl;
-    Assert( d_equalityEngine.consistent() );
-    d_sentLemma = false;
-    d_addedFact = false;
-    d_full_check_incomplete = false;
-    d_set_eqc.clear();
-    d_set_eqc_list.clear();
-    d_eqc_emptyset.clear();
-    d_eqc_univset.clear();
-    d_eqc_singleton.clear();
-    d_congruent.clear();
-    d_nvar_sets.clear();
-    d_var_set.clear();
-    d_most_common_type.clear();
-    d_most_common_type_term.clear();
-    d_pol_mems[0].clear();
-    d_pol_mems[1].clear();
-    d_members_index.clear();
-    d_singleton_index.clear();
-    d_bop_index.clear();
-    d_op_list.clear();
-    d_card_enabled = false;
-    d_t_card_enabled.clear();
-    d_rels_enabled = false;
-    d_eqc_to_card_term.clear();
-
-    std::vector< Node > lemmas;
```

Comment:
Just to confirm. This local variable `lemmas` is replaced with `InferenceManager::d_pendingLemmas`. Right?

Reply:
Yes, that's right. It's easier this way since we don't have to pass that local variable around everywhere.